traffic_ops/traffic_ops_golang/deliveryservice/health.go

```diff
@@ -115,13 +116,40 @@ func getMonitorHealth(tx *sql.Tx, ds tc.DeliveryServiceName, monitorFQDNs []stri
 
 // addHealth adds the given cache states to the given data and totals, and returns the new data and totals
 func addHealth(ds tc.DeliveryServiceName, data map[tc.CacheGroupName]tc.HealthDataCacheGroup, totalOnline uint64, totalOffline uint64, crStates tc.CRStates, crConfig tc.CRConfig) (map[tc.CacheGroupName]tc.HealthDataCacheGroup, uint64, uint64) {
+
+	var deliveryService tc.CRConfigDeliveryService
+	var ok bool
+	var topology string
+	var cacheGroupNameMap = make(map[string]bool)
+	var cacheCapabilities = make(map[string]bool)
+	var skip bool
+
+	if deliveryService, ok = crConfig.DeliveryServices[string(ds)]; !ok {
+		log.Errorln("delivery service not found in CRConfig")
+		return map[tc.CacheGroupName]tc.HealthDataCacheGroup{}, 0, 0
+	}
+
+	if deliveryService.Topology != nil {
+		topology = *deliveryService.Topology
+		if topology != "" {
+			if top, ok := crConfig.Topologies[topology]; !ok {
+				log.Errorf("CRConfig topologies does not contain DS topology: %s", topology)
```

Comment:
Should this return an error instead of logging this and returning data as though the DS's topology *were* found?